control-plane/agents/src/common/errors.rs

```diff
@@ -1018,6 +1030,24 @@ impl From<SvcError> for ReplyError {
                 source,
                 extra,
             },
+            SvcError::InsufficientHealthyReplicas { .. } => ReplyError {
+                kind: ReplyErrorKind::FailedPrecondition,
+                resource: ResourceKind::VolumeSnapshot,
+                source,
+                extra,
+            },
+            SvcError::SnapshotMaxLimit { .. } => ReplyError {
+                kind: ReplyErrorKind::ResourceExhausted,
```

Comment:
Wonder if this is the correct error code here, could be confused with no free space in the pool for taking the snapshot?

Reply:
/// The system is not in a state required for the operation's execution.
    FailedPrecondition = 9,
    
    /// Operation was attempted past the valid range.
    OutOfRange = 11,
    
    /// Some resource has been exhausted.
    ResourceExhausted = 8,
    
     /// The caller does not have permission to execute the specified operation.
    PermissionDenied = 7,

@tiagolobocastro These 4 seem to be a candidate. How about `OutOfRange` ?

Reply:
`PermissionDenied` seems to be more reasonable for me.

Reply:
I'm also leaning more towards `OutOfRange`

Reply:
okay, let me keep `OutOfRange` then.